sdk/go/common/resource/properties.go

```diff
@@ -84,7 +84,10 @@ type Computed struct {
 // encountered, it means the resource has not yet been created, and so the output value is unavailable.  Note that an
 // output property is a special case of computed, but carries additional semantic meaning.
 type Output struct {
-	Element PropertyValue // the eventual value (type) of the output property.
+	Element      PropertyValue // the value of this output if it is resolved.
+	Known        bool          `json:"-"` // true if this output's value is known.
+	Secret       bool          `json:"-"` // true if this output's value is secret.
+	Dependencies []URN         `json:"-"` // the dependencies associated with this output.
```

Comment:
The `json` tags are interesting--what's the idea here?

Reply:
I didn't originally have the `json` tags, but found that we have this test:

https://github.com/pulumi/pulumi/blob/7ebac460a3c9ca1c2d6625f31e17297985d47ab9/pkg/resource/stack/deployment_test.go#L333

Which fails otherwise because it isn't expecting the new fields.